omegatrade-app/frontend/src/app/components/stock-dashboard/stock-dashboard.component.ts

```diff
@@ -0,0 +1,218 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Chart, StockChart } from 'angular-highcharts';
+import { ActivatedRoute } from '@angular/router';
+import { RestService } from '../../services/rest.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { interval, Subscription } from "rxjs";
+import { take } from 'rxjs/operators';
+import { SnackBarService } from '../../services/snackbar.service';
+
+@Component({
+  selector: 'app-stock-dashboard',
+  templateUrl: './stock-dashboard.component.html',
+  styleUrls: ['./stock-dashboard.component.css']
+})
+export class StockDashboardComponent implements OnInit , OnDestroy{
+
+  selectedCompany: string = "";
+  lastUpdatedTime;
+  stock: StockChart;
+  companies: any;
+  loader: boolean = false;
+  subscription: Subscription;
+  timerIds = [];
+  
+  constructor(private snackBarService: SnackBarService, private router: ActivatedRoute, private _snackBar: MatSnackBar, private restService: RestService) {
+  }
+
+  /**
+   *  Function to Initiate component.
+   *  Initiating company lists
+   */
+  ngOnInit(): void {
+    this.selectedCompany = this.router.snapshot.queryParamMap.get('companyId');
+    this.getCompanies();
+  }
+
+  /**
+   *  Function to get all companies.
+   *  @returns  {null}
+   */
+  getCompanies() {
+    this.restService.getData('companies/list')
+      .pipe(take(1))
```

Comment:
If I understand this correctly, it will only use the first company that is returned by the function `companies/list` function. If so, I think it would be good to modify the `companies/list` function to accept an optional parameter `limit` that will limit the number of rows that is returned from the database.

Reply:
`companies/list` will return all the companies from the database. `.pipe(take(1))` is used to avoid potential memory leaks. The async pipe subscribes to an Observable and returns the latest value it has emitted. and unsubscribes automatically when the component gets destroyed